src/consensus/vote.cpp

```diff
@@ -114,6 +121,15 @@ void VoteManager::retreieveVotes_() {
   for (auto const& v : verified_votes) {
     auto pbft_round = v.getRound();
     auto hash = v.getHash();
+    
+    //Rebroadcast our own next votes in case we were partitioned...
+    if (v.getVoterAddr() == node_addr_ && v.getStep() > FIRST_FINISH_STEP && db_->getPbftMgrField(PbftMgrRoundStep::PbftStep) > EXTENDED_PARTITION_STEPS) {
```

Comment:
v.getStep() >= FIRST_FINISH_STEP, should include step 4

Reply:
Ok fixed.